### PR TITLE
feat(drift2): HZN-1559,HZN-1570,HZN-1571

### PR DESCRIPTION
**Pull request by mattixtech**
Drift 2.0 API Enhancements Frontend Work

This PR updates opennms-js to provide access to the updated flow API changes in Opennms.

All of the updates have been manually tested.

Jira issues:
https://issues.opennms.org/browse/HZN-1571
https://issues.opennms.org/browse/HZN-1570
https://issues.opennms.org/browse/HZN-1559

### CODE REVIEW

File: src/api/ServerMetadata.ts [hunk 1]
```diff
@@ -100,11 +100,21 @@ export class ServerMetadata {
     }
   }
 
-  /** Returs a convenient data structure with all capabilities listed. */
+  /** Does this version support the drift 2.0 flows enhancements? */
+  public enhancedFlows() {
+    if (this.type && this.type === ServerTypes.MERIDIAN) {
+      return this.version.ge('2020.0.0');
```
Comment (mattixtech) [inline src/api/ServerMetadata.ts]: I made an assumption on the version here,

Comment (RangerRick) [reply src/api/ServerMetadata.ts]: We're not releasing 2019 until the newer Drift stuff is in, AFAIK.  @j-white  ?

Comment (j-white) [reply src/api/ServerMetadata.ts]: Yep, it should be 2019

Comment (mattixtech) [reply src/api/ServerMetadata.ts]: Ok, fixed it to 2019.

File: src/dao/FlowDAO.ts [hunk 2]
```diff
@@ -161,31 +193,304 @@ export class FlowDAO extends BaseDAO {
...
      * @param end - the end of the timespan to query (defaults to now)
      * @param step - the requested time interval between rows
+     * @param includeOther - include an additional "other" result that
+     *                       represents everything that does not match the top N
      * @param exporterNodeCriteria - the node ID or foreignSource:foreignId tuple
      * @param ifIndex - filter for flows that came through this SNMP interface
      */
     public async getSeriesForTopNConversations(N?: number, start?: number, end?: number,
-                                               step?: number, exporterNodeCriteria?: string,
+                                               step?: number, includeOther?: boolean, exporterNodeCriteria?: string,
```
[the remark this answers is not in the document]
Comment (mattixtech) [reply src/dao/FlowDAO.ts]: Good catch! Thanks. Addressed in latest commit.

File: src/dao/FlowDAO.ts [hunk 3]
```diff
@@ -598,3 +604,8 @@ export class FlowDAO extends BaseDAO {
         }
     }
 }
+
+export interface ITopNOptions {
```
[the remark this answers is not in the document]
Comment (mattixtech) [reply src/dao/FlowDAO.ts]: Yep, this is a cleaner way of doing it. Thanks. Fixed in latest commit.

[the remark this answers is not in the document]
Comment (mattixtech) [reply src/dao/FlowDAO.ts]: I've only updated the API methods for conversations that we previously identified as breaking backwards compatibility. We can look into updating the other methods in the future for consistency.

**Pull request merged**